tensorflow/stream_executor/rocm/rocm_event.h

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_STREAM_EXECUTOR_ROCM_ROCM_EVENT_H_
+#define TENSORFLOW_STREAM_EXECUTOR_ROCM_ROCM_EVENT_H_
+
+#include "tensorflow/stream_executor/event.h"
+#include "tensorflow/stream_executor/lib/status.h"
+#include "tensorflow/stream_executor/rocm/rocm_driver.h"
+#include "tensorflow/stream_executor/rocm/rocm_stream.h"
+
+namespace stream_executor {
+namespace rocm {
+
+// ROCMEvent wraps a hipEvent_t in the platform-independent EventInterface
+// interface.
+class ROCMEvent : public internal::EventInterface {
```

Comment:
If the GPUDriver suggested by me is implemented, ROCMEvent seems to be identical to CUDAEvent, thus there will be further de-duplications.

Can you also investigate how many other places can be de-duplicated?